Azure-Samples/azure-search-openai-demo
# Add support for an optional login and document level access control system.

## Purpose

* Add an optional login and document level access control system
* Support for access control based on the User Object ID and Group ID
* Add optional login UI to the frontend components
* Add optional authentication tokens and authentication understanding capabilities to the backend
* Use security filters in Azure Cognitive Search to implement the access control system.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

* Test the code
1. Follow the manual setup instructions in `LoginAndAclSetup.md`.
1. After running `azd up` or running locally, a login button will appear at the top right. Use this to log into your account in your Azure AD tenant.
1. Enable the optional `oids` or `groups` security filters in the developer settings. Sample access control can be setup by setting `AZURE_ADLS_GEN2_STORAGE_ACCOUNT` to a valid data lake storage account prior to running `azd up` or running `prepdocs.py`. Access control values can also be manually managed using the `manageacl.ps1` script.

## Other Information
